### PR TITLE
Introducing controller-gen to generate CRDs

### PR DESCRIPTION
**Pull request by ricardomaraschini**
Prior to this PR all changes needed to be manually executed in the CRDs yaml file. This PR introduces the use of `controller-gen` to generate, from the Go types, proper CRD definitions. 

This has been split into two commits, the first one breaks the Makefile into multiple shell scripts placed in the `hack/` directory (following a well established default) while the second adds comments to the Go code and CRDs.

### CODE REVIEW

File: hack/get-proto.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script installs a pinned version of protoc and protoc-gen-go binaries
+# under output/protoc directory.
+
+PROTO_VER=3.15.8
+GEN_GO_VER=v1
+PROTO_OUT_DIR=$(pwd)/output/protoc
```
Comment (otaviof) [inline hack/get-proto.sh:22]: I think you're looking for `${PWD}` instead.

**Review by otaviof — COMMENTED** [review]
Very cool initiative. I recommend all scripts have the following:

```bash
#!/usr/bin/env bash

set -eu -o pipefail
```

File: hack/get-proto.sh (hunk 1 above)
Comment (ricardomaraschini) [reply hack/get-proto.sh]: done.

File: hack/run-kuttl.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This scripts run the kuttl e2e tests. Kuttl binary is expected to exists
+# under output/kuttl directory (as set by get-kuttl.sh script).
+
+output/kuttl/kuttl test --timeout=180 e2e
```
[the remark this answers is not in the document]
Comment (ricardomaraschini) [reply hack/run-kuttl.sh]: done.

**Comment by ricardomaraschini** [comment]
> Very cool initiative. I recommend all scripts have the following:
> 
> ```shell
> #!/usr/bin/env bash
> 
> set -eu -o pipefail
> ```

done.

**Comment by ricardomaraschini** [comment]
Can we get this in ?

**Comment by otaviof** [comment]
Ohh, we don't have bot integration on this repository. I'm okay on merging the changes without the bot's intervention 👍

**Pull request closed without merging**